Source/WebKit/UIProcess/API/Cocoa/_WKWebPushDaemonConnection.h

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (C) 2024 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+
+#import <WebKit/WKFoundation.h>
+
+typedef NS_ENUM(NSInteger, _WKWebPushPermissionState) {
+    _WKWebPushPermissionStateDenied,
+    _WKWebPushPermissionStateGranted,
+    _WKWebPushPermissionStatePrompt,
+};
+
+@class _WKWebPushMessage;
+@class _WKWebPushSubscriptionData;
+
+NS_ASSUME_NONNULL_BEGIN
+
+WK_EXTERN
+@interface _WKWebPushDaemonConnectionConfiguration : NSObject
+
+- (instancetype)init;
+@property (nonatomic, copy) NSString *machServiceName;
+@property (nonatomic, copy) NSString *partition;
+
+@end
+
+WK_EXTERN
+@interface _WKWebPushDaemonConnection : NSObject
+
++ (instancetype)new NS_UNAVAILABLE;
+- (instancetype)init NS_UNAVAILABLE;
+- (instancetype)initWithConfiguration:(_WKWebPushDaemonConnectionConfiguration *)configuration;
+
+- (void)getPushPermissionStateForScope:(NSURL *)scopeURL completionHandler:(void (^)(_WKWebPushPermissionState))completionHandler;
```

Comment:
I think this should be named `getPushPermissionStateForOrigin:` since the granularity of permission is an origin and I don't think we have any plans on changing that.